### DISCUSSION TITLE
`flip` doesn't use default fallback placements when called inside custom middleware

### DISCUSSION THREAD

**Discussion by eternalthinker** [floating-ui/floating-ui#2919] [Q&A] [answered]
I'm trying to run `flip` in a custom middleware conditionally. The minimal version is as follows:
```js
{
  name: "customFlip",
  fn: (state) => {
    return flip().fn(state);
  }
}
```

However, this doesn't flip a `bottom-start` placement to `top-start`. Instead, the alignment is switched to `bottom-end`.
I have to provide an explicit fallbackPlacements array `['top-start']` for this to work.
Using `flip()` directly in the middleware array works as expected. 
Demo: https://codepen.io/eternalthinkerr/pen/YzbNKPj

![fui-flip](https://github.com/floating-ui/floating-ui/assets/2195458/1674bedb-bd5e-475a-8527-fc92acac08f5)

Why doesn't the default options work when flip is called this way? Am I missing something?

**Comment by atomiks** [comment, marked as answer]
`flip()` relies on its name `"flip"` to work. So the custom wrapping one needs to be `name: "flip"` not `name: "customFlip"`
